File: Java/src/main/java/Solution801_900/Solution896.java
package Solution801_900;

/**
 * @author windows
 */
public class Solution896 {
    public boolean isMonotonic(int[] nums) {
        boolean up = true;
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                up = false;
                break;
            }
        }
        boolean down = true;
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] < nums[i + 1]) {
                down = false;
                break;
            }
        }
        return down || up;
    }


    public static void main(String[] args) {
        Solution896 solution896 = new Solution896();
    }
}

File: C++/src/Solution801_900/Solution896.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution896.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool Solution::isMonotonic(vector<int> &nums) {
    int n = nums.size();
    if (nums[0] < nums[n - 1]){
        for (int i = 0; i < n - 1; i++){
            if (nums[i] > nums[i + 1]){
                return false;
            }
        }
        return true;
    }else{
        for (int i = 0; i < n - 1; i++){
            if (nums[i] < nums[i + 1]){
                return false;
            }
        }
        return true;
    }
}
